pkg/policy/mapstate.go

```diff
@@ -242,7 +247,7 @@ func (keys MapState) RemoveDependent(owner Key, dependent Key) {
 	}
 }
 
-// MergeReferences adds owners and dependents from entry 'entry' to 'e'. 'entry' is not modified.
+// MergeReferences adds owners, dependents, and DerivedFromRules from entry 'entry' to 'e'. 'entry' is not modified.
```

Comment:
how does this merging of the labels impact production environments? I'm thinking from two perspectives:
1. Debuggability. This feels like it might potentially improve the ability to debug map state changes, but I'm not quite sure how a developer or user might do that.
2. Performance. This seems like a whole bunch of extra string manipulation and potentially allocations. If this is called often, it could impact policy generation times.

Reply:
1. Logging now contains this, also there is `GetRealizedPolicyRuleLabelsForKey()`, but that seems to be used for unit testing only, currently?

2. Addressed this by adding a new commit that exports `labels.MergeSorted()` that now merges in place and minimises memory allocations.